src/main/java/io/xpring/xrpl/DefaultXpringClient.java

```diff
@@ -0,0 +1,153 @@
+package io.xpring.xrpl;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.xpring.proto.*;
+import io.xpring.proto.XRPLedgerAPIGrpc.XRPLedgerAPIBlockingStub;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigInteger;
+import java.util.Objects;
+
+/**
+ * A client that can submit transactions to the XRP Ledger.
+ *
+ * @see "https://xrpl.org"
+ */
+public class DefaultXpringClient implements XpringClientDecorator {
+    // TODO: Use TLS!
+    public static final String XPRING_TECH_GRPC_URL = "grpc.xpring.tech:80";
+
+    // A margin to pad the current ledger sequence with when submitting transactions.
+    private static final int LEDGER_SEQUENCE_MARGIN = 10;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    // Channel is the abstraction to connect to a service endpoint
+    private final XRPLedgerAPIBlockingStub stub;
+
+    /**
+     * No-args Constructor.
+     */
+    public DefaultXpringClient() {
+        this(ManagedChannelBuilder
+            .forTarget(XPRING_TECH_GRPC_URL)
+            // Let's use plaintext communication because we don't have certs
+            // TODO: Use TLS!
+            .usePlaintext(true)
+            .build()
+        );
+    }
+
+    /**
+     * Required-args Constructor, currently for testing.
+     *
+     * @param channel A {@link ManagedChannel}.
+     */
+    DefaultXpringClient(final ManagedChannel channel) {
+        // It is up to the client to determine whether to block the call. Here we create a blocking stub, but an async
+        // stub, or an async stub with Future are always possible.
+        this.stub = XRPLedgerAPIGrpc.newBlockingStub(channel);
+    }
+
+    /**
+     * Get the balance of the specified account on the XRP Ledger.
+     *
+     * @param xrplAccountAddress The X-Address to retrieve the balance for.
+     * @return A {@link BigInteger} with the number of drops in this account.
+     * @throws XpringKitException If the given inputs were invalid.
+     */
+    public BigInteger getBalance(final String xrplAccountAddress) throws XpringKitException {
+        if (!Utils.isValidXAddress(xrplAccountAddress)) {
+            throw XpringKitException.xAddressRequiredException;
+        }
+
+        Objects.requireNonNull(xrplAccountAddress, "xrplAccountAddress must not be null");
```

Comment:
Would `null` pass in `Utils.isValidXAddress`? If not then you wouldn't need this line.